csharp/src/Ice/Communicator.cs

```diff
@@ -463,31 +463,26 @@ private class ObserverUpdater : Instrumentation.IObserverUpdater
                 ClientAcm = new Acm(this, "Ice.ACM.Client", new Acm(this, "Ice.ACM", Acm.ClientDefault));
                 ServerAcm = new Acm(this, "Ice.ACM.Server", new Acm(this, "Ice.ACM", Acm.ServerDefault));
 
+                int frameSizeMax = GetPropertyAsByteSize("Ice.MessageSizeMax") ?? 1024 * 1024;
+                if (frameSizeMax < 1 || frameSizeMax > 0x7fffffff / 1024)
```

Comment:
I'm not following what this check is for. Should we should just replace `0x7fffffff` by `int.MaxValue`?

Reply:
Pretty sure this check isn't right as frameSizeMax is now in bytes

Reply:
Right I think this was just to ensure that KB conversion to bytes didn't overflow, I will fix it

Reply:
We have this check in a few other places too.

Reply:
fixed